Artifacts/windows-deploymentgroup/InstallAndConfigureDeploymentGroupAgent.ps1

```diff
@@ -144,6 +144,76 @@ function Configure-DeploymentGroupAgent
 
     if ($runAsAutoLogon)
     {
+        # Create a PS session for the user to trigger the creation of the registry entries required for autologon
+        if ($windowsLogonAccount.Split("\").Count -eq 2)
+        {
+            $domain = $windowsLogonAccount.Split("\")[0]
+        }
+        else
+        {
+            $domain = Hostname
+        }
+        $computerName = "localhost"
+        $password = ConvertTo-SecureString $windowsLogonPassword -AsPlainText -Force
+        $credentials = New-Object System.Management.Automation.PSCredential("$domain\\$windowsLogonAccount", $password)
+        Enter-PSSession -ComputerName $computerName -Credential $credentials
+        Exit-PSSession
+
+        $ErrorActionPreference = "stop"
+
+        # 120 seconds timeout
```

Comment:
Move `$timeout` definition closer to the `while` loop.

Reply:
done